### PR TITLE
bug 1762859 - Silently ignore add(0) on counter and labeled_counter

### PR DESCRIPTION
**Pull request by chutten**

### CODE REVIEW

**Comment by chutten** [comment]
(I'm guessing one or two of the language bindings probably have their own tests that will fail that I missed by only testing Rust and Python locally, but we'll see)

**Review by travis79 — APPROVED** [review]
This matches what I expect from the discussion in the Bug. r+

File: docs/user/reference/metrics/counter.md [hunk 1]
```diff
@@ -123,8 +124,8 @@ Glean.controls.refreshPressed.add(5);
 
 ### `testGetValue`
 
-Gets the recorded value for a given counter metric.  
-Returns the count if data is stored.  
+Gets the recorded value for a given counter metric.
+Returns the count if data is stored.
```
Comment (badboy) [inline docs/user/reference/metrics/counter.md:128]: Note that the double-spacing at line-end here was load-bearing: It enforces line breaks in the generated HTML.

File: glean-core/src/metrics/counter.rs [hunk 2]
```diff
@@ -62,16 +63,23 @@ impl CounterMetric {
...
-            );
-            return;
-        }
+        match amount.cmp(&0) {
+            Ordering::Less => {
+                record_error(
+                    glean,
+                    &self.meta,
+                    ErrorType::InvalidValue,
+                    format!("Added negative or zero value {}", amount),
```
Comment (badboy) [inline glean-core/src/metrics/counter.rs:72]:
```suggestion
                    format!("Added negative value {}", amount),
```

**Review by badboy — CHANGES_REQUESTED** [review]

**Pull request merged**